### PR TITLE
Add JSON lexer

### PR DESCRIPTION
**Pull request by fabon-f**

### CODE REVIEW

File: src/noir/lexers/json.cr [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require "../lexer"
+class Noir::Lexers::JSON < Noir::Lexer
+  tag "json"
+  filenames %w(*.json)
+  mimetypes %w(application/json application/vnd.api+json application/hal+json)
+
+  state :root do
+    rule /\s+/, Text::Whitespace
+    rule /"/, Str::Double, :string
+    rule /true|false|null/, Keyword::Constant
```
Comment (makenowjust) [inline src/noir/lexers/json.cr]: I prefer to wrap `\b` around keywords for correctness.

**Comment by makenowjust** [comment]
It works well in respect of JSON following RFC 8259 exactly. I think, however, syntax highlighters should work permissively. For example `tsconfig.json` accept JS like comments. What do you think?

File: src/noir/lexers/json.cr (hunk 1 above)
Comment (fabon-f) [reply src/noir/lexers/json.cr]: I fixed it.

File: src/noir/lexers/json.cr [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require "../lexer"
+class Noir::Lexers::JSON < Noir::Lexer
+  tag "json"
+  filenames %w(*.json)
+  mimetypes %w(application/json application/vnd.api+json application/hal+json)
+
+  state :root do
+    rule /\s+/, Text::Whitespace
+    rule /"/, Str::Double, :string
+    rule /(true|false|null)\b/, Keyword::Constant
```
Comment (makenowjust) [inline src/noir/lexers/json.cr]: Add `\b` to both sides. For instance it matches `"true"` on `"xtrue"`.

**Review by makenowjust — CHANGES_REQUESTED** [review]

**Review by makenowjust — APPROVED** [review]

**Comment by makenowjust** [comment]
Sorry, closing this PR is my mistaken.

I surveyed JSON language definition of some syntax highlighters, and there is no highlighter to accept comment. (However VSCode has language definition for JSON with comments.) So, it is ready to merge now.

Thank you @fabon-f for contribution.

**Pull request merged**